linkify.js

```diff
@@ -15,10 +15,33 @@ function redirectGoLink(details) {
     return {redirectUrl: local_storage_cache[golink].url}
 }
 
+/**
+ * Extract the go link from the search link and redirect to the user's stored website if exists.
+ */
+function redirectSearchLink(details) {
+    let golink = details.url.replace(/(.*)\:\/\/(.*)=go%2F/, "")
+    golink = golink.replace(/%2F/, "/");
+
+    // Remove trailing query string content of the search link
+    // Yahoo: &fr=opensearch, Ecosia: &addon=opensearch
+    golink = golink.replace(/\&fr=opensearch|\&addon=opensearch/, "")
+
+    if (!local_storage_cache[golink]) {
+        return;
+    }
+    return {redirectUrl: local_storage_cache[golink].url}
+}
+
 browser.storage.onChanged.addListener(() => getLocalStorageData());
 
 browser.webRequest.onBeforeRequest.addListener(
     redirectGoLink,
     { "urls": ["*://go/*"] },
     ["blocking"],
 );
+
+browser.webRequest.onBeforeRequest.addListener(
+    redirectSearchLink,
+    { "urls": ["*://*/*=go%2F*"] },
```

Comment:
Thought we were hard-coding search engines in here. Please replace `*` by specific search engines.